demos/menu-surface.html

```diff
@@ -171,6 +183,43 @@
           menu.open = !menu.open;
         });
 
+        var changeAnchorElement = document.querySelector('input[name="anchor-element"]');
+        var hoistMenu = document.querySelector('input[name="hoist-menu"]');
+        var fixedPosition = document.querySelector('input[name="fixed-position"]');
+
+        changeAnchorElement.addEventListener('click', function() {
+          if (changeAnchorElement.checked) {
+            menu.setMenuSurfaceAnchorElement(document.getElementById('overflow-icon'));
+            menu.hoistMenuToBody();
+            hoistMenu.checked = true;
+            hoistMenu.disabled = true;
+            fixedPosition.disabled = true;
+          } else {
+            menu.setMenuSurfaceAnchorElement(menuButtonEl.parentElement);
+            hoistMenu.disabled = false;
+            fixedPosition.disabled = false;
+          }
+        });
+
+        hoistMenu.addEventListener('change', function() {
+          if (hoistMenu.checked) {
+            menu.hoistMenuToBody();
+          } else {
+            var temp = menuEl.parentElement.removeChild(menuEl);
+            menuButtonEl.parentElement.appendChild(temp);
+            menu = mdc.menuSurface.MDCMenuSurface.attachTo(menuEl);
+            // Trigger events to ensure the new menu is setup to reflect the configurations selected.
+            var event = new Event('change');
+            document.querySelector('input[name="anchor-position"]:checked').dispatchEvent(event);
```

Comment:
are these to call this method https://github.com/material-components/material-components-web/pull/3067/files#diff-553c5b418ac1d319c3b4d8ff2a42902bR280?